integration/container/stop_test.go

```diff
@@ -42,6 +43,47 @@ func TestStopContainerWithRestartPolicyAlways(t *testing.T) {
 	}
 }
 
+// TestStopContainerWithTimeout checks that ContainerStop with
+// a timeout works as documented, i.e. in case of negative timeout
+// waiting is not limited (issue #35311).
+func TestStopContainerWithTimeout(t *testing.T) {
+	t.Parallel()
+	defer setupTest(t)()
+	client := request.NewAPIClient(t)
+	ctx := context.Background()
+
+	testCmd := container.WithCmd("sh", "-c", "sleep 2 && exit 42")
+	testData := []struct {
+		timeout, expectedExitCode int
+	}{
+		// In case container is forcefully killed, 137 is returned,
+		// otherwise the exit code from the above script
+		{0, 137}, // zero timeout: sleep won't finish, container is killed
+		{1, 137}, // too small timeout: same as above
+		{3, 42},  // big enough timeout for normal container stop
+		{-1, 42}, // unlimited timeout: same as above
+	}
+
+	for _, d := range testData {
+		name := container.WithName(t.Name() + "_" + strconv.Itoa(d.timeout))
```

Comment:
Use "t.Run" so the different cases run as separate tests. This makes it easiee to know which case failed (if there is a failure)